src/smt/set_defaults.cpp

```diff
@@ -553,6 +552,12 @@ void setDefaults(LogicInfo& logic, bool isInternalSubsolver)
       log.enableIntegers();
       log.arithOnlyLinear();
     }
+    if (!logic.areIntegersUsed())
```

Comment:
Alternatively, we could just delete line 554 which sets that arithmetic should be only linear, right?  I guess this would by default make QF_S -> QF_SNIA?  If this is the case, I believe I agree with this PR.

Reply:
I thought about that. If you are fine with switching from `QF_SLIA` to `QF_SNIA`, this would probably be better. Not only is it cleaner, it also avoids problem when we enable transcendentals (which are also disabled by `arithOnlyLinear`...)